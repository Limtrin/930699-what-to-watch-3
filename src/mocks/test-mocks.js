export const filmsList = [
  {
    id: `shutter-island`,
    name: `Shutter Island`,
    genre: `detective`,
    year: 2010,
    image: `img/shutter-island.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  },
  {
    id: `snatch`,
    name: `Snatch`,
    genre: `comedy`,
    year: 2009,
    image: `img/snatch.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  }
];

export const film = {
  id: `war-of-the-worlds`,
  name: `War of the Worlds`,
  genre: `action`,
  year: 2008,
  image: `img/war-of-the-worlds.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `2h 22m`,
  rating: 9.5,
  votes: 123,
  director: `Frank Spaskukotski`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
  reviews: [
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    }
  ],
  starring: [
    `Tim Robbins`,
    `Morgan Freeman`,
    `Bob Gunton`,
    `William Sadler`,
    `Clancy Brown`,
    `Gil Bellows`,
    `James Whitmore`
  ],
};
