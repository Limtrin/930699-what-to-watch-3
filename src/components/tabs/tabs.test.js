import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs.jsx";
import {film} from "../../mocks/test-mocks.js";

it(`Should <Tabs/> component render`, () => {
  const tree = renderer.create(<Tabs film={film} />).toJSON();

  expect(tree).toMatchSnapshot();
});
