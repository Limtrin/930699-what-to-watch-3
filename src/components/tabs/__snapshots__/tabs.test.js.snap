// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should <Tabs/> component render 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      9.5
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        123
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Frank Spaskukotski
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tim Robbins, Morgan Freeman, Bob Gunton, William Sadler
         and other
      </strong>
    </p>
  </div>
</div>
`;
